routingtable/vrf/vrf_registry.go

```diff
@@ -9,14 +9,16 @@ var globalRegistry *vrfRegistry
 
 func init() {
 	globalRegistry = &vrfRegistry{
-		vrfs: make(map[string]*VRF),
+		vrfsName: make(map[string]*VRF),
+		vrfsID:   make(map[uint32]*VRF),
 	}
 }
 
 // vrfRegistry holds a reference to all active VRFs. Every VRF have to have a different name.
 type vrfRegistry struct {
-	vrfs map[string]*VRF
-	mu   sync.Mutex
+	vrfsName map[string]*VRF
```

Comment:
Why do we track the vrfs by booth name and id? Could we just use a single map[id]->VRF and check the name of the VRF with a given id.

Reply:
but then we can't lookup a vrf by name.
I could do it the other way round to do the second lookup with `map[name]id` but if we wanna lookup by id in the future this could also lead to some problems.
Memory wise i don't think that's a big difference, since we are only storing `map[string] pointer to to VRF`

Reply:
OK. Good arguments.